internal/objectstore/dynamic_cache.go

```diff
@@ -70,15 +70,41 @@ type DynamicCache struct {
 	updateFns       []store.UpdateFn
 	updateMu        sync.Mutex
 
+	timeoutFunc     func(context.Context, store.Key, chan bool)
+	waitForSyncFunc func(context.Context, store.Key, *DynamicCache, informers.GenericInformer, <-chan struct{}, chan bool)
+
 	useDynamicClient bool
 }
 
+func timeout(ctx context.Context, key store.Key, done chan bool) {
```

Comment:
Is there a better name or comment for this? Timeout is going to trip someone up in the future.